activerecord/lib/active_record/associations/preloader/association.rb

```diff
@@ -6,7 +6,7 @@ class Preloader
       class Association #:nodoc:
         def initialize(klass, owners, reflection, preload_scope)
           @klass         = klass
-          @owners        = owners
+          @owners        = owners.uniq(&:object_id)
```

Comment:
Use `__id__` just in case. Ref #38990

```suggestion
          @owners        = owners.uniq(&: __id__)
```